### PR TITLE
[JENKINS-71553] Add history widget factory for job not implementing `Queue.Task`

### PR DESCRIPTION
**Pull request by Vlatombe**
## [JENKINS-71553] Restore missing building history for external jobs

Follows up #7932. What was missed previously is that unlike other Job types, external job does NOT implement `Queue.Task` (as it is not running within Jenkins, it has no queue items). Still, it has a history widget, but it can't contain any queue item.

See [JENKINS-71553](https://issues.jenkins.io/browse/JENKINS-71553).

### Testing done

Confirmed with interactive testing that the external job history that was not visible in 2.409 is now visible with this change.

### Proposed changelog entries

- Restore missing build history for external jobs (regression in 2.409).

### Proposed upgrade guidelines

N/A

```[tasklist]
### Submitter checklist
- [X] The Jira issue, if it exists, is well-described.
- [X] The changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developers, depending on the change) and are in the imperative mood (see [examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)). Fill in the **Proposed upgrade guidelines** section only if there are breaking changes or changes that may require extra steps from users during upgrade.
- [x] There is automated testing or an explanation as to why this change has no tests.
- [x] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadocs, as appropriate.
- [x] New deprecations are annotated with `@Deprecated(since = "TODO")` or `@Deprecated(forRemoval = true, since = "TODO")`, if applicable.
- [x] New or substantially changed JavaScript is not defined inline and does not call `eval` to ease future introduction of Content Security Policy (CSP) directives (see [documentation](https://www.jenkins.io/doc/developer/security/csp/)).
- [x] For dependency updates, there are links to external changelogs and, if possible, full differentials.
- [x] For new APIs and extension points, there is a link to at least one consumer.
```

### Desired reviewers

@MarkEWaite

Before the changes are marked as `ready-for-merge`:

```[tasklist]
### Maintainer checklist
- [x] There are at least two (2) approvals for the pull request and no outstanding requests for change.
- [x] Conversations in the pull request are over, or it is explicit that a reviewer is not blocking the change.
- [x] Changelog entries in the pull request title and/or **Proposed changelog entries** are accurate, human-readable, and in the imperative mood.
- [x] Proper changelog labels are set so that the changelog can be generated automatically.
- [x] If the change needs additional upgrade steps from users, the `upgrade-guide-needed` label is set and there is a **Proposed upgrade guidelines** section in the pull request title (see [example](https://github.com/jenkinsci/jenkins/pull/4387)).
- [x] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).
```

### CODE REVIEW

**Review by NotMyFault — APPROVED** [review]
Thanks!

**Review by timja — APPROVED** [review]

**Comment by timja** [comment]
/label ready-for-merge

----

This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

**Review by MarkEWaite — APPROVED** [review]
Thanks very much @Vlatombe !  Tested interactively and confirmed that the problem is resolved.  Build history is visible for the external job that I had defined.

**Comment by MarkEWaite** [comment]
I'm accelerating the merge of this pull request so that it will be included in the weekly release that is delivered tomorrow.

**Pull request merged**
